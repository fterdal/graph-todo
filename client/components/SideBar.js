import React from 'react';
import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';

const SideBar = (props) => {
  const welcome = !props.data.me ? (<div />)
                                 : (<div>Welcome, {props.data.me.email}</div>);
  // console.log('props.data:', props.data);
  return (
    <div className="sidebar-container">
      <div />
      {welcome}
      <a className="button auth-button">Login</a>
      <a className="button auth-button">Signup</a>
    </div>
  )
}

const meQuery = gql`
  query {
    me {
      id
      email
    }
  }
`;

export default compose(
  graphql(meQuery),
)(SideBar)
