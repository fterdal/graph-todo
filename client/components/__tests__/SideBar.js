import React from 'react';
import { SideBar } from '../SideBar';
import renderer from 'react-test-renderer';

test('Sidebar renders when logged in', () => {
  // const notLoggedIn = { me: null };
  const loggedIn = {
    me: {
      id: 7,
      email: 'hello@world.com',
    }
  };
  const sidebar = renderer.create(
    <SideBar data={loggedIn} />,
  );
  let sidebarTree = sidebar.toJSON();
  expect(sidebarTree).toMatchSnapshot();
});

xtest('Sidebar renders when not logged in', () => {
  const notLoggedIn = { me: null };
  const sidebar = renderer.create(
    <SideBar data={notLoggedIn} />,
  );
  let sidebarTree = sidebar.toJSON();
  expect(sidebarTree).toMatchSnapshot();
});
