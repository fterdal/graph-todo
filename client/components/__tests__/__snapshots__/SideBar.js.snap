// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders when logged in 1`] = `
<div
  className="sidebar-container"
>
  <div />
  <div>
    Welcome, 
    hello@world.com
  </div>
  <a
    className="button auth-button"
  >
    Account
  </a>
  <a
    className="button auth-button"
    onClick={[Function]}
  >
    Logout
  </a>
</div>
`;
